src/mmw/apps/bigcz/clients/cuahsi/search.py

```diff
@@ -102,25 +111,38 @@ def parse_records(series, services):
 def group_series_by_location(series):
     """
     Group series catalog results by location. Each result contains fields
-    common across all variables.
+    common across all variables, and some aggregations.
 
     Ref: https://github.com/WikiWatershed/model-my-watershed/issues/1931
     """
-    result = {}
+    groups = {}
     for record in series:
-        location = record['location']
-        if location not in result:
-            result[location] = {
-                'ServCode': record['ServCode'],
-                'ServURL': record['ServURL'],
-                'Sitename': record['Sitename'],
-                'latitude': record['latitude'],
-                'location': record['location'],
-                'longitude': record['longitude'],
-                'beginDate': parse_date(record['beginDate']),
-                'endDate': parse_date(record['endDate']),
-            }
-    return result.values()
+        group = groups.get(record['location'])
+        if not group:
+            groups[record['location']] = [record]
+        else:
+            group.append(record)
+
+    records = []
+    for location, group in groups.iteritems():
+        records.append({
+            'ServCode': group[0]['ServCode'],
+            'ServURL': group[0]['ServURL'],
+            'location': group[0]['location'],
+            'Sitename': group[0]['Sitename'],
+            'latitude': group[0]['latitude'],
+            'longitude': group[0]['longitude'],
+            'sample_mediums': ', '.join(sorted(set([r['samplemedium']
```

Comment:
If we're going to normalize the aggregate attribute names into snakey_case, we could consider normalizing all the attribute names